src/CMakeLists.txt

```diff
@@ -0,0 +1,133 @@
+set(CMAKE_C_FLAGS_DEBUG "-DUSE_ASSERT_CHECKING=1")
+
+if (UNIX)
+  set(CMAKE_SHARED_LINKER_FLAGS "-L${PG_LIBDIR}")
+  set(CMAKE_MODULE_LINKER_FLAGS "-L${PG_LIBDIR}")
+  set(CMAKE_C_FLAGS "${PG_CFLAGS}")
+  set(CMAKE_CPP_FLAGS "${PG_CPPFLAGS}")
+  set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -g")
```

Comment:
Are we sure it safe/correct to add debugging to our so even if the main postgres lib was compiled w/o debug symbols?

Reply:
I don't know any reason why it would be a problem. Do you?

Reply:
So there is nothing to be defined as USE flag looks like, it makes it even simplier, what I was asking especially is something like OpenCV package which has OCV_OPTIONs to be enabled/disable to support different kind of features (gpu, protocols, codecs support), for reference:
https://github.com/opencv/opencv/blob/master/CMakeLists.txt

Once merged I will produce happily Gentoo ebuild and request to put it into official repos :-)

Reply:
@cevian - compilation with -g could/should be option (ON or OFF - I think better to have OFF by default)